### PR TITLE
PR for branch basic-ui that adds basic User Interface

### PR DESCRIPTION
**Pull request by jkanyi-web**
- Created file app.rb that defines methods for managing a library system, allowing users to list books and people, create new people (students or teachers), create new books, create rentals, and list rentals associated with a specific person, with interactions based on user input.
- Created file main.rb that defines a set of functions for handling user interactions in a library management application. It includes functions for listing books and people, creating new people (students or teachers), creating new books, creating rentals, listing rentals for a person, and quitting the application. It also has a main loop that displays a menu and handles user choices, interacting with an instance of the App class defined in the app.rb file.

### CODE REVIEW

File: app.rb [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+  end
+
+  def list_people
+    Person.all.each do |person|
+      puts "Name: #{person.correct_name}"
+      puts "Age: #{person.age}"
+      puts "Classroom: #{person.classroom&.label}" if person.is_a?(Student)
+      puts "Specialization: #{person.specialization}" if person.is_a?(Teacher)
+      puts '----------'
+    end
```
Comment (usorfaitheloho) [inline app.rb]: Hello, @jkanyi-web ,
Good job coming this far with your project 🚀 
Please kindly fix this error in your "person.rb" file. Your code isn't running as a result of the error encountered on line 4.
Kindly fix all errors shown in your console before submitting for a review again 👍 

Console error:
image

**Review by usorfaitheloho — CHANGES_REQUESTED** [review]
> Hi @jkanyi-web  ,
> 
> Good job so far! There are some issues that you still need to work on to go to the next project but you are almost there!

  [ ](https://camo.githubusercontent.com/679757134a8fee76cfe1941f9f3ecb0e0cf56dd02da4498268ce31f0dacdfca3/68747470733a2f2f6d65646961302e67697068792e636f6d2f6d656469612f556461535559436c723932636a57706556782f67697068792e676966)
> 
> ## Highlights
> * Built interactive console app ✔️
> * Descriptive pull request ✔️
> * Used GitHub flow ✔️
> 
> ## Required Changes ♻️
> _Check the comments under the review._
> 
> ## Optional suggestions
> _Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._
> 
> Cheers and Happy coding!👏👏👏
> 
> Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Please, remember to tag me in your question so I can receive the notification.
> 
> **_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**
> 
> _As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app.rb (hunk 1 above)
Comment (DeliceLydia) [reply app.rb]: Addressed

File: main.rb [hunk 2]
```diff
@@ -1,13 +1,136 @@
...
+def list_rentals_for_person(app)
+  puts 'Enter person ID:'
+  person_id = gets.chomp.to_i
+  app.list_rentals_for_person(person_id)
+end
+
+def handle_quit
+  puts 'Thanks for using the app, Goodbye!'
+  exit
+end
```
Comment (DeliceLydia) [inline main.rb]: ## Optional
- [x] I highly suggest that you fix number 7 before merging I am getting an error while quitting
Screenshot 2023-11-10 at 14 14 52

**Review by DeliceLydia — APPROVED** [review]
## Hi @jkanyi-web, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
# Congratulations! 🎉 

## Highlights
- Addressed changes suggested by the previous reviewer



### Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**